### PR TITLE
Replace meterpreter scripts that have replacement post modules

### PR DESCRIPTION
**Pull request by bcook-r7**
This adds a translation layer so that post modules are automatically run when a user refers to a legacy meterpreter script. It also deletes the meterpreter scripts that have replacements. See #7789

This also fixes an issue noted by @wwebb-r7 where all modules that specified InitialAutoRunScripts that referred to Meterpreter scripts were slightly broken due to the migrate meterpreter script being broken.

```
'InitialAutoRunScript' => 'migrate -f'
```

This automatically converts them to use the corresponding post module.

## Verification

- [x] Validate that modules that use the InitialAutoRunScript referenced above actually migrate.
- [x] Verify that meterpreter modules are replaced with post modules automatically.

### CODE REVIEW

**Comment by OJ** [comment]
You're a wizard Harry!

**Comment by busterb** [comment]
I'm not a wizard, Hagrid, I'm just Harry!

File: lib/msf/base/sessions/scriptable.rb [hunk 1]
```diff
@@ -52,12 +52,56 @@ def execute_file
...
+
+  #
   # Executes the supplied script, Post module, or local Exploit module with
   #   arguments +args+
   #
   # Will search the script path.
   #
   def execute_script(script_name, *args)
+    post_module = legacy_script_to_post_module(script_name)
+    script_name == post_module if !post_module.nil?
```
Comment (wvu-r7) [inline lib/msf/base/sessions/scriptable.rb]: Did you mean to make this an assignment?

Comment (busterb) [inline lib/msf/base/sessions/scriptable.rb]: no

Comment (wvu-r7) [inline lib/msf/base/sessions/scriptable.rb]: Yes!

**Comment by wvu-r7** [comment]
# Release Notes

This PR removes Meterpreter scripts for which there are post modules, adding a check that resolves a script name to its corresponding module reference name.

**Pull request merged**